### PR TITLE
2140: Foreground push notifications

### PR DESCRIPTION
**Pull request by LeandraH**
### Short description

Letting push notifications appear in the foreground too.

### Proposed changes

- Installed [Notifee](https://github.com/invertase/notifee) which seems quite well-supported
- Used that to show a notification in the foreground if one comes in while the app is open
- Added a notification icon for Android (it needs to be black-and-white, and with a transparent background)

### Side effects

Should be nothing

### Notes for testing

I have not been able to receive notifications in emulated iOS, only on my real iPhone.

On Android, I have only been able to test on emulated Android, it would be cool if we could have at least one test on a real Android device.

And in both cases, the test CMS takes too long until the notification is in the news, so you see the error screen for a good second or two before the actual content appears. This seems to not happen in the real CMS though.

### Resolved issues

Fixes: #2140 

---

### CODE REVIEW

**Review by steffenkleinle — COMMENTED** [review]
Tested on android, great job!

File: native/src/utils/PushNotificationsManager.ts [hunk 1]
```diff
@@ -106,18 +107,32 @@ export const useForegroundPushNotificationListener = ({
...
+            id: 'integreat',
+            name: 'Integreat Channel',
+            importance: AndroidImportance.HIGH,
+          })
+
+          await notifee.displayNotification({
+            title: message.notification.title,
+            body: message.notification.body,
+            android: {
+              smallIcon: 'ic_notification',
```
[the remark this answers is not in the document]
Comment (LeandraH) [reply native/src/utils/PushNotificationsManager.ts]: Thanks, I thought they don't have push notifications enabled. I was wrong, they do.
The ic stands for icon, it's Android's default naming, I changed the name.

**Review by f1sh1918 — CHANGES_REQUESTED** [review]
Push notification pops up nicely but when i click on it the content can not be found. After a few seconds it refreshes and forwards to the correct news page. 

Tested on android and ios real device

**Comment by LeandraH** [comment]
@f1sh1918 The test CMS takes too long until the notification is in the news, so you see the error screen for a good second or two before the actual content appears. This seems to not happen in the real CMS though.

**Comment by f1sh1918** [comment]
> @f1sh1918 The test CMS takes too long until the notification is in the news, so you see the error screen for a good second or two before the actual content appears. This seems to not happen in the real CMS though.

hm i don't think this is the issue i get also an error

```
The 'navigation' object hasn't been initialized yet. This might happen if you don't have a navigator mounted, or if the navigator hasn't finished mounting. See https://reactnavigation.org/docs/navigating-without-navigation-prop#handling-initialization for more details.
```

**Comment by f1sh1918** [comment]
Hm i really have no idea how to fix this. Pushed some time into it.
I mean if the cms needs some time to get the push notification in the api. Shouldn't it help to `setTimeout` around the `notifee.displayNotification` to just wait until we display it instead of timeout the `onForegroundEvent` 
Nevertheless even more timeout doesn't help
I even checked the `onReady` function of the `NavigationContainer` and passed a prop to check if its ready which seem to be the case.
I also tried to set the `ref` but looks like i can not do it in combination with `linking` (deeplinking)

**Review by f1sh1918 — APPROVED** [review]
Ok i put some useless effort into it to fix it for test cms which i couldn't get work and then i realized that this issue really does not happen in production cms.
So sorry to block this @LeandraH and @steffenkleinle 
I'm even wondering why even a `setTimeout()` surrounding `notifee.displayNotification` does not help for test cms , but no matter

**Comment by LeandraH** [comment]
You're right, it's more than just the CMS taking some time. I'm currently trying to handle the tapping the same way as background notification is handled, by opening the redirect route but that screen somehow never opens. It's a weird issue. I'll keep trying 🤷

**Comment by f1sh1918** [comment]
> You're right, it's more than just the CMS taking some time. I'm currently trying to handle the tapping the same way as background notification is handled, by opening the redirect route but that screen somehow never opens. It's a weird issue. I'll keep trying 🤷

hm maybe the solution is good enough. Even i think this timeout is not very stable :/

**Comment by LeandraH** [comment]
Oh. All good, thank you! But it's frustrating to not know why it doesn't work with the Test CMS

**Comment by LeandraH** [comment]
I just tested removing the timeout fully and using the real CMS, and it did show the error message. So it looks like something like the timeout is still necessary 😞

**Pull request merged**